subprojects/jacoco/src/integTest/groovy/org/gradle/testing/jacoco/plugins/rules/JacocoPluginCoverageVerificationIntegrationTest.groovy

```diff
@@ -0,0 +1,352 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.testing.jacoco.plugins.rules
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.testing.jacoco.plugins.fixtures.JavaProjectUnderTest
+import org.gradle.util.Requires
+import spock.lang.Unroll
+
+import static JacocoViolationRulesLimit.Insufficient
+import static JacocoViolationRulesLimit.Sufficient
+import static org.gradle.util.TestPrecondition.FIX_TO_WORK_ON_JAVA9
+
+@Requires(FIX_TO_WORK_ON_JAVA9)
+class JacocoPluginCoverageVerificationIntegrationTest extends AbstractIntegrationSpec {
+
+    private final JavaProjectUnderTest javaProjectUnderTest = new JavaProjectUnderTest(testDirectory)
+    private final static String[] TEST_TASK_PATH = [':test'] as String[]
+    private final static String[] JACOCO_COVERAGE_VERIFICATION_TASK_PATH = [':jacocoTestCoverageVerification'] as String[]
+    private final static String[] TEST_AND_JACOCO_COVERAGE_VERIFICATION_TASK_PATHS = TEST_TASK_PATH + JACOCO_COVERAGE_VERIFICATION_TASK_PATH
+    private final static String[] INTEG_TEST_AND_JACOCO_COVERAGE_VERIFICATION_TASK_PATHS = [':integrationTest', ':jacocoIntegrationTestCoverageVerification'] as String[]
+
+    def setup() {
+        javaProjectUnderTest.writeBuildScript().writeSourceFiles()
+    }
+
```

Comment:
I think JaCoCo will simply check both rules with the same thresholds. So we just do extra work. I am going to add a test case for it.